reindexer/src/main/scala/uk/ac/wellcome/platform/reindexer/services/ReindexService.scala

```diff
@@ -0,0 +1,41 @@
+package uk.ac.wellcome.platform.reindexer.services
+
+import javax.inject.Inject
+
+import com.amazonaws.services.dynamodbv2.model.PutItemResult
+import com.twitter.inject.Logging
+import uk.ac.wellcome.models.{Reindexable, Transformable}
+import uk.ac.wellcome.platform.reindexer.models.ReindexAttempt
+import uk.ac.wellcome.utils.GlobalExecutionContext.context
+
+import scala.concurrent.Future
+
+class ReindexService[T <: Transformable with Reindexable[
+  String]] @Inject()(
+  reindexTrackerService: ReindexTrackerService,
+  reindexTargetService: ReindexTargetService[T])
+    extends Logging {
+
+  def run: Future[PutItemResult] =
+    for {
+      indices <- reindexTrackerService.getIndicesForReindex
+      attempt = indices.map(ReindexAttempt(_, Nil, 0))
+      _ <- attempt.map(processReindexAttempt).get
```

Comment:
What happens if `indices` is a `None`? I _think_ that attempt will be `None` as well so this line will throw a `None.get` exception... is this true? is there a test checking what happens in this case?